security/bouncycastle-fips/src/test/java/io/quarkus/ts/security/bouncycastle/fips/BouncyCastleFipsIT.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.ts.security.bouncycastle.fips;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.core.IsEqual.equalTo;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class BouncyCastleFipsIT {
+
+    @QuarkusApplication
+    static RestService app = new RestService();
+
+    @Test
+    public void testListProviders() {
+        given()
+                .when()
+                .get("/api/listProviders")
+                .then()
+                .statusCode(200)
+                .body(containsString("BCFIPS"));
+    }
+
+    @Test
+    public void testSHA256withRSAandMGF1() {
```

Comment:
how is SHA256withRSAandMGF1 special that it's checked in fips module?

`testSHA256withRSAandMGF1Availability` maybe sounds better